src/argus/notificationprofile/media/email.py

```diff
@@ -80,27 +80,30 @@ def validate(cls, instance: RequestDestinationConfigSerializer, email_dict: dict
 
         return form.cleaned_data
 
-    @staticmethod
-    def is_deletable(destination: DestinationConfig) -> Union[str, NoneType]:
+    @classmethod
+    def raise_if_not_deletable(cls, destination: DestinationConfig) -> Union[str, NoneType]:
         """
-        Returns None if the given email destination is able to be deleted
-        and an error message if it was defined by an outside source or
-        is in use by any notification profiles
+        Raises a NotDeletableError if the given email destination is not able
+        to be deleted (if it was defined by an outside source or is in use by
+        any notification profiles)
         """
         if destination.settings["synced"]:
-            return "Cannot delete this email destination since it was defined by an outside source."
+            raise cls.NotDeletableError(
+                "Cannot delete this email destination since it was defined by an outside source."
+            )
 
         connected_profiles = destination.notification_profiles.all()
         if connected_profiles:
             profiles = ", ".join([str(profile) for profile in connected_profiles])
-            return "".join(
-                [
-                    "Cannot delete this destination since it is in use in the notification profile(s): ",
-                    profiles,
-                    ".",
-                ]
+            raise cls.NotDeletableError(
+                "".join(
+                    [
+                        "Cannot delete this destination since it is in use in the notification profile(s): ",
+                        profiles,
+                        ".",
+                    ]
+                )
```

Comment:
Nitpick: This join operation looks absurdly unreadable for its simple intended purpose.

```suggestion
            raise cls.NotDeletableError(
                f"Cannot delete this destination since it is in use in the notification profile(s): {profiles}."
                )
```